### PR TITLE
return matched path segments from path.Tree.Find()

### LINKED ISSUE #94: A new variant for path.Tree.Find() is needed
Basepaths will need to be stripped from the request path. When matching for basepaths, the result should included the matched part of the actual request path for correct trimming. Otherwise, any basepath that include `*` will not be working properly.

### PR DESCRIPTION
**Pull request by rockspore**
Fix #94

### CODE REVIEW

**Comment by theganyo** [comment]
Tests for the path match results?

File: path/tree_test.go [hunk 1]
```diff
@@ -78,10 +78,13 @@ func TestWildcardTree(t *testing.T) {
 	}
 	for _, test := range find {
 		path := strings.Split(test.path, "/")
-		got := tree.Find(path, 0)
+		got, matched := tree.Find(path, 0)
 		if test.value != got {
 			t.Errorf("for: %v, want: %v, got: %v", test.path, test.value, got)
 		}
+		if s := strings.Join(matched, "/"); s != test.path {
```
Comment (rockspore) [inline path/tree_test.go:85]: This is the added check to test the returned matchedPath. There are checks added in other tests too.

**Comment by rockspore** [comment]
> Tests for the path match results?

Please see the comment I made.

**Comment by theganyo** [comment]
Oops. Of course.

**Pull request closed without merging**